### PR TITLE
Enhancement to allow different WebSocket applications to be registered on different ports.

### PR DESCRIPTION
**Pull request by spericas**
Enhancement to allow different WebSocket applications to be registered on different ports. Due to a restriction in Tyrus, you cannot share the same path even if registered on different ports, so this is a restriction on this enhancement. WebSocket unit tests re-structured to use @HelidonTest. For more info see issue #5643 and `WebSocketEndpointAppTest`.

### CODE REVIEW

**Review by tjquinno — COMMENTED** [review]
A couple possible changes.

File: microprofile/websocket/src/main/java/io/helidon/microprofile/tyrus/WebSocketCdiExtension.java [hunk 1]
```diff
@@ -148,62 +148,61 @@ private void registerWebSockets() {
...
-                endpointConfigs.forEach(wsCfg -> wsRoutingBuilder.endpoint(rootPath, wsCfg));
+            Set<Class<? extends ServerApplicationConfig>> appClasses = app.applicationClasses();
 
-                // Create routing wsRoutingBuilder
-                addWsRouting(wsRoutingBuilder.build(), namedRouting, routingNameRequired, c.getName());
-            } else {
+            if (appClasses.isEmpty()) {
                 // Direct registration without calling application class
+                Optional<String> contextRoot = Optional.empty();
+                String rootPath = contextRoot.orElse(DEFAULT_WEBSOCKET_PATH);
```
[the remark this answers is not in the document]
Comment (spericas) [reply microprofile/websocket/src/main/java/io/helidon/microprofile/tyrus/WebSocketCdiExtension.java:163]: That's a bug after separation of the old code, will fix

File: microprofile/websocket/src/main/java/io/helidon/microprofile/tyrus/WebSocketApplication.java [hunk 2]
```diff
@@ -52,12 +52,21 @@ public static Builder builder() {
...
+        return applicationClasses.isEmpty() ? Optional.empty() : Optional.of(applicationClasses.iterator().next());
+    }
+
+    /**
+     * Get access to all application classes. Possibly an empty set.
+     *
+     * @return Set of application classes.
+     */
+    public Set<Class<? extends ServerApplicationConfig>> applicationClasses() {
+        return applicationClasses;
```
[the remark this answers is not in the document]
Comment (spericas) [reply microprofile/websocket/src/main/java/io/helidon/microprofile/tyrus/WebSocketApplication.java:69]: Sounds like a good suggestion

[the remark this answers is not in the document]
Comment (tjquinno) [reply microprofile/websocket/src/main/java/io/helidon/microprofile/tyrus/WebSocketApplication.java]: Or return an `UnmodifiableSet` wrapper around it.

**Review by tjquinno — APPROVED** [review]

**Pull request merged**